client/src/components/categorical/category.js

```diff
@@ -164,6 +170,11 @@ class Category extends React.Component {
       false
     );
 
+    const truncatedString = maybeTruncateString(
+      metadataField,
+      globals.categoryDisplayStringMaxLength
+    );
+
     return (
```

Comment:
I am OK if we defer to a future PR, but this JSX is out of control -- super hard to understand.  It needs refactoring, either by pulling out bits into separate functions, or sub-components (if reusable).     (ditto for value.js)

Reply:
Value.js is being addressed in #1209, but I agree, annotations have increased the complexity immensely.  I'll write up an issue for the category component and will take the learnings for #1209 and apply them here.